File: app/src/main/java/ru/voronezhtsev/weatherapp/models/data/network/weather/WeatherResponse.java

package ru.voronezhtsev.weatherapp.models.data.network.weather;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

//todo Прочитать нужны ли setter's в Gson
public class WeatherResponse {
    @SerializedName("weatherInfo")
    @Expose
    private List<Weather> weather = null;
    @SerializedName("main")
    @Expose
    private Main main;
    @SerializedName("name")
    @Expose
    private String name;

    public List<Weather> getWeather() {
        return weather;
    }

    public void setWeather(List<Weather> weather) {
        this.weather = weather;
    }

    public Main getMain() {
        return main;
    }

    public String getName() {
        return name;
    }
}

File: app/src/main/java/ru/voronezhtsev/weatherapp/models/data/network/WeatherResponse.kt
package ru.voronezhtsev.weatherapp.models.data.network

import com.google.gson.annotations.SerializedName

data class WeatherResponse(@SerializedName("id") val id: Long,
                           @SerializedName("main") val main: Main,
                           @SerializedName("weather") val weather: List<Weather>,
                           @SerializedName("name") val name: String,
                           @SerializedName("dt") val dateTime: Long
)

data class Weather(@SerializedName("id") val id: Int,
                   @SerializedName("icon") val icon: String,
                   @SerializedName("description") val description: String
)